fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -59,11 +59,6 @@ def self.run(params)
           build_number = live_version.build.version
         end
 
-        # Remove leading zeros from version string (eg. 1.02 -> 1.2)
-        if version
-          version = version.split(".").map(&:to_i).join(".")
-        end
```

Comment:
Note: This was added here https://github.com/fastlane/fastlane/pull/15235

> The download_dsyms action fails when an app's version has a leading zero in one of its parts, such as a version "4.07". AppStore reports this version as "4.7", so the action fails to find its dSYM files, since "4.07" != "4.7".

According to the previous PR, App Store Connect (not the public API) seemed to return the version without leading zeros previously. So it makes sense to change this to follow the latest spec of APIs.

Reply:
Thank you for the review @ainame!

Just to add, I couldn't find any official Apple API docs on how the leading zeros are being treated so it's merely based on the observed behaviour of the Apple API. It may break (again) in the future but I guess it shouldn't affect too many people as not many people use leading zeros in their versions 😉